### PR TITLE
Tolerate existing paging control in SearchWithPaging

### PR DESCRIPTION
**Pull request by stevekuznetsov**
With the current implementation of `Search`, our interaction with this library has been as such: 
 - gather knowledge in order to fully qualify an LDAP query
 - create a `SearchRequest`
 - get a `SearchResponse`
 - unpack the response for salient information

The current API for `SearchWithPaging` does not honor this workflow and expects something *other* than a fully-qualified `SearchRequest` to function. This PR adds another API call that allows for a fully qualified search request to be all that is necessary to issue a request to search with client-side page stitching.

@johnweldon @liggitt  PTAL

### CODE REVIEW

File: search.go [hunk 1]
```diff
@@ -268,6 +268,9 @@ func NewSearchRequest(
 	}
 }
 
+// SearchWithPaging augments the given search request with a paging control containing the given paging size, then
+// issues a search with buffered paging using the request. All paged responses are buffered so that the result can
+// be returned atomically.
 func (l *Conn) SearchWithPaging(searchRequest *SearchRequest, pagingSize uint32) (*SearchResult, error) {
```
Comment (liggitt) [inline search.go]: This function is really doing two things:
1. adding the paging control (which is easy to do yourself)
2. searching/paging/appending (which is hard to do yourself)

What if we made this tolerate search requests that already had a paging control set? That would let us avoid expanding the interface surface area

cases:
no paging control present, adds one matching pagingSize
paging control matching pagingSize present, skips adding a control
paging control not matching pagingSize present, returns error

Comment (stevekuznetsov) [inline search.go]: What would I pass through in the third case? `SearchWithPaging(mySearchRequest, someIgnoredInt)` ?

Comment (liggitt) [inline search.go]: you'd pass a pageSize that matched the already-present control's PagedSize

**Comment by stevekuznetsov** [comment]
@liggitt Changed `SearchWithPaging` to have the desired function and added some tests.

**Comment by liggitt** [comment]
LGTM. @johnweldon, any feedback on tolerating an already-present paging control like this?

**Comment by johnweldon** [comment]
Looks like a nice improvement, and low risk.  LGTM :shipit:

**Pull request merged**